### ISSUE TITLE
clang-10 bug `parser end of file` (reproducible, not upstream, mingw only)

### ISSUE THREAD

**Issue by Pro** [msys2/MINGW-packages#6576]
I'm not sure if this is completely a MINGW issue or an issue with clang itself.

msys2 Version: msys2 v20200602.0.0

In our Azure DevOps Pipeline we are doing a CI crosscompile build of the source with current MingW and clang 10.0.0

It always results in the following error: `<eof> parser at end of file`

For full log see here:
https://dev.azure.com/open62541/open62541/_build/results?buildId=1834&view=logs&j=ad9e6d9e-6e5f-5424-e835-306608605e5b&t=3de98930-df77-56d3-5f2c-a58644372cc4&l=465

```
[ 16%] Building C object CMakeFiles/open62541-object.dir/src/pubsub/ua_pubsub_manager.c.obj
[ 16%] Building C object CMakeFiles/open62541-object.dir/src/pubsub/ua_pubsub_ns0.c.obj
cmake.exe : Stack dump:
At D:\a\1\s\tools\azure-devops\win\build.ps1:117 char:5
+     & cmake --build . --config Debug
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (Stack dump::String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
0.	Program arguments: C:\msys64\mingw64\bin\clang.exe --target=x86_64-w64-mingw32 -DUA_ARCHITECTURE_WIN32 
-DUA_DYNAMIC_LINKING_EXPORT @CMakeFiles/open62541-object.dir/includes_C.rsp -std=c99 -pipe -Wall -Wextra -Wpedantic 
-Werror -Wno-static-in-inline -Wno-overlength-strings -Wno-unused-parameter -Wc++-compat -Wformat -Wformat-security 
-Wformat-nonliteral -Wmissing-prototypes -Wstrict-prototypes -Wredundant-decls -Wuninitialized -Winit-self -Wcast-qual 
-Wstrict-overflow -Wnested-externs -Wmultichar -Wundef -fno-strict-aliasing -fexceptions -g -o 
CMakeFiles\open62541-object.dir\src\pubsub\ua_pubsub_ns0.c.obj -c D:\a\1\s\src\pubsub\ua_pubsub_ns0.c 
1.	<eof> parser at end of file
2.	Code generation
 
#0 0x000000000278f4b8 (C:\msys64\mingw64\bin\clang.exe+0x238f4b8)
 #1 0x0000000002795e7c (C:\msys64\mingw64\bin\clang.exe+0x2395e7c)
 #2 0x00000000027a4002 (C:\msys64\mingw64\bin\clang.exe+0x23a4002)
 #3 0x00000000027a8378 (C:\msys64\mingw64\bin\clang.exe+0x23a8378)
 #4 0x000000000277f8d8 (C:\msys64\mingw64\bin\clang.exe+0x237f8d8)
 #5 0x000000000179c2bc (C:\msys64\mingw64\bin\clang.exe+0x139c2bc)
 #6 0x00000000017a58a0 (C:\msys64\mingw64\bin\clang.exe+0x13a58a0)
 #7 0x0000000001f83fc1 (C:\msys64\mingw64\bin\clang.exe+0x1b83fc1)
 #8 0x00000000047a1e09 (C:\msys64\mingw64\bin\clang.exe+0x43a1e09)
 #9 0x00000000033b7919 (C:\msys64\mingw64\bin\clang.exe+0x2fb7919)
#10 0x0000000002520569 (C:\msys64\mingw64\bin\clang.exe+0x2120569)
#11 0x00000000024e51d1 (C:\msys64\mingw64\bin\clang.exe+0x20e51d1)
#12 0x00000000025d3339 (C:\msys64\mingw64\bin\clang.exe+0x21d3339)
#13 0x000000000040269f (C:\msys64\mingw64\bin\clang.exe+0x269f)
#14 0x000000000040197a (C:\msys64\mingw64\bin\clang.exe+0x197a)
#15 0x00000000023c95b9 (C:\msys64\mingw64\bin\clang.exe+0x1fc95b9)
#16 0x0000000001d208cf (C:\msys64\mingw64\bin\clang.exe+0x19208cf)
#17 0x00000000023c9855 (C:\msys64\mingw64\bin\clang.exe+0x1fc9855)
#18 0x00000000023a36da (C:\msys64\mingw64\bin\clang.exe+
0x1fa36da
)
#19 0x00000000023a3fdb (C:\msys64\mingw64\bin\clang.exe+0x1fa3fdb)
#20 0x00000000023adf51 (
C:\msys64\mingw64\bin\clang.exe
+0x1fadf51)
#21 0x0000000004a9c9e9 (C:\msys64\mingw64\bin\clang.exe+0x469c9e9)
#22 0x00000000004013c4 (C:\msys64\mingw64\bin\clang.exe+
0x13c4
)
#23 0x000000000040151b (C:\msys64\mingw64\bin\clang.exe+0x151b)
#24 0x00007ffb00dc7974 (C:\windows\System32\KERNEL32.DLL+0x17974)
#25 0x00007ffb036ca271 (C:\windows\SYSTEM32\ntdll.dll+0x6a271)
clang
: error: clang frontend command failed due to signal (use -v to see invocation)
clang version 10.0.0 (https://github.com/msys2/MINGW-packages.git 3f880aaba91a3d9cdfb222dc270274731a2119a9)
Target: x86_64-w64-windows-gnu
Thread model: posix
InstalledDir: C:\msys64\mingw64\bin
clang
: note: diagnostic msg: PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace, 
preprocessed source, and associated run script.
clang
: note: diagnostic msg: 
********************
PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang: note: diagnostic msg: C:\Users\VSSADM~1\AppData\Local\Temp\ua_pubsub_ns0-94b9da.c
clang
: note: diagnostic msg: C:\Users\VSSADM~1\AppData\Local\Temp\ua_pubsub_ns0-94b9da.sh
clang: note: diagnostic msg: 
********************
mingw32-make[2]: *** [CMakeFiles\open62541-object.dir\build.make:478: 
CMakeFiles/open62541-object.dir/src/pubsub/ua_pubsub_ns0.c.obj] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:526: CMakeFiles/open62541-object.dir/all] Error 2
mingw32-make: *** [Makefile:171: all] Error 2


*** Make failed. Exiting ... ***
```

I cannot reproduce this issue on my local msys2 installation with the same source code.

It's also not so easy to get the preprocessed source, since it's running on Azure DevOps.

Do you have any suggestions how I can further debug the issue?

**Comment by mati865** [comment]
Azure Pipelines has this stupid default Git configuration causing it to convert LF to CRLF when cloning the repo but Clang should handle it just fine.
Looks like upstream bug but without preprocessed source and run script it's even hard to guess what could cause the crash.

**Comment by Pro** [comment]
@mati865 thanks for the feedback!
I was able to hack a file upload for Azure Pipelines to get the preprocessed source and run script:


This contains the preprocessed source and run script:
[PipelineFiles.zip (2.7MB)](https://github.com/msys2/MINGW-packages/files/4750537/PipelineFiles.zip)

This contains the full repo checkout from `D:/a/1/s`:

[repo.zip (10MB)](https://github.com/msys2/MINGW-packages/files/4750548/repo.zip)


Anything more I can do to help investigate the issue?

I think that this could be related to limited RAM on the Azure Pipeline and somehow clang runs OOM.

----

If one is interested, I used the https://transfer.sh service to get the files:

```powershell

        function transfer.sh ($filename)
        {
            $file = Get-Item $filename;
            (Invoke-WebRequest -Method PUT -InFile $file.FullName -Uri https://transfer.sh/$($file.Name)).Content
        }

        Get-ChildItem -Path C:\Users\VSSADM~1\AppData\Local\Temp\*.sh, C:\Users\VSSADM~1\AppData\Local\Temp\*.c |
            Compress-Archive -DestinationPath C:\Users\VSSADM~1\AppData\Local\Temp\PipelineFiles.zip
        $url = (transfer.sh C:\Users\VSSADM~1\AppData\Local\Temp\PipelineFiles.zip)
        Write-Host -ForegroundColor Red "`n`n*** Transferred pipeline files: $url ***"

        Compress-Archive -Path D:/a/1/s -DestinationPath C:\Users\VSSADM~1\AppData\Local\Temp\repo.zip
        $url = (transfer.sh C:\Users\VSSADM~1\AppData\Local\Temp\repo.zip)
        Write-Host -ForegroundColor Red "`n`n*** Transferred repo files: $url ***"

```

**Comment by Pro** [comment]
Ok, I was able to reproduce the issue locally in my Windows 10 VM at least partially:

1. Install msys2 (v20200602.0.0) from (https://github.com/msys2/msys2-installer/releases/download/2020-06-02/msys2-x86_64-20200602.exe)
2. Install required packages:
```
pacman -S mingw-w64-x86_64-toolchain
pacman -S mingw-w64-x86_64-clang
```
3. Download the zip with preprocessed source and run script and extract it: [PipelineFiles.zip (2.7MB)](https://github.com/msys2/MINGW-packages/files/4750537/PipelineFiles.zip)
4. `chmod +x ./ua_pubsub_ns0-c752d2.sh`
5. `./ua_pubsub_ns0-c752d2.sh`

The interesting part now is, that the issue is not deterministic.

You need to run the script multiple times, and at some point the script crashes:

```
fancy@win-vm MSYS /c/Users/fancy/Downloads/PipelineFiles
$ ./ua_pubsub_ns0-c752d2.sh

fancy@win-vm MSYS /c/Users/fancy/Downloads/PipelineFiles
$ ./ua_pubsub_ns0-c752d2.sh

fancy@win-vm MSYS /c/Users/fancy/Downloads/PipelineFiles
$ ./ua_pubsub_ns0-c752d2.sh
Stack dump:
0.      Program arguments: C:\msys64\mingw64\bin\clang.exe -cc1 -triple x86_64-w64-windows-gnu -emit-obj -mrelax-all -disable-free -disable-llvm-verifier -discard-value-names -main-file-name ua_pubsub_ns0.c -mrelocation-model pic -pic-level 2 -mthread-model posix -mframe-pointer=none -relaxed-aliasing -fmath-errno -fno-rounding-math -masm-verbose -mconstructor-aliases -munwind-tables -target-cpu x86-64 -dwarf-column-info -debug-info-kind=limited -dwarf-version=4 -debugger-tuning=gdb -D UA_ARCHITECTURE_WIN32 -D UA_DYNAMIC_LINKING_EXPORT -Wall -Wextra -Wpedantic -Werror -Wno-static-in-inline -Wno-overlength-strings -Wno-unused-parameter -Wc++-compat -Wformat -Wformat-security -Wformat-nonliteral -Wmissing-prototypes -Wstrict-prototypes -Wredundant-decls -Wuninitialized -Winit-self -Wcast-qual -Wstrict-overflow -Wnested-externs -Wmultichar -Wundef -std=c99 -ferror-limit 19 -fmessage-length 0 -fno-use-cxa-atexit -fgnuc-version=4.2.1 -fobjc-runtime=gcc -fexceptions -fseh-exceptions -fdiagnostics-show-option -faddrsig -x c ua_pubsub_ns0-c752d2.c
1.      <eof> parser at end of file
2.      Code generation
 #0 0x000000000278f4b8 (C:\msys64\mingw64\bin\clang.exe+0x238f4b8)
 #1 0x0000000002795e7c (C:\msys64\mingw64\bin\clang.exe+0x2395e7c)
 #2 0x00000000027a4002 (C:\msys64\mingw64\bin\clang.exe+0x23a4002)
 #3 0x00000000027a8378 (C:\msys64\mingw64\bin\clang.exe+0x23a8378)
 #4 0x000000000277f8d8 (C:\msys64\mingw64\bin\clang.exe+0x237f8d8)
 #5 0x000000000179c2bc (C:\msys64\mingw64\bin\clang.exe+0x139c2bc)
 #6 0x00000000017a58a0 (C:\msys64\mingw64\bin\clang.exe+0x13a58a0)
 #7 0x0000000001f83fc1 (C:\msys64\mingw64\bin\clang.exe+0x1b83fc1)
 #8 0x00000000047a1e09 (C:\msys64\mingw64\bin\clang.exe+0x43a1e09)
 #9 0x00000000033b7919 (C:\msys64\mingw64\bin\clang.exe+0x2fb7919)
#10 0x0000000002520569 (C:\msys64\mingw64\bin\clang.exe+0x2120569)
#11 0x00000000024e51d1 (C:\msys64\mingw64\bin\clang.exe+0x20e51d1)
#12 0x00000000025d3339 (C:\msys64\mingw64\bin\clang.exe+0x21d3339)
#13 0x000000000040269f (C:\msys64\mingw64\bin\clang.exe+0x269f)
#14 0x000000000040197a (C:\msys64\mingw64\bin\clang.exe+0x197a)
#15 0x0000000004a9b01e (C:\msys64\mingw64\bin\clang.exe+0x469b01e)
#16 0x00000000004013c4 (C:\msys64\mingw64\bin\clang.exe+0x13c4)
#17 0x000000000040151b (C:\msys64\mingw64\bin\clang.exe+0x151b)
#18 0x00007ffbdd537bd4 (C:\WINDOWS\System32\KERNEL32.DLL+0x17bd4)
#19 0x00007ffbdeb4ce51 (C:\WINDOWS\SYSTEM32\ntdll.dll+0x6ce51)
./ua_pubsub_ns0-c752d2.sh: Zeile 4:  1172 Segmentation fault      "C:\\msys64\\mingw64\\bin\\clang.exe" "-cc1" "-triple" "x86_64-w64-windows-gnu" "-emit-obj" "-mrelax-all" "-disable-free" "-disable-llvm-verifier" "-discard-value-names" "-main-file-name" "ua_pubsub_ns0.c" "-mrelocation-model" "pic" "-pic-level" "2" "-mthread-model" "posix" "-mframe-pointer=none" "-relaxed-aliasing" "-fmath-errno" "-fno-rounding-math" "-masm-verbose" "-mconstructor-aliases" "-munwind-tables" "-target-cpu" "x86-64" "-dwarf-column-info" "-debug-info-kind=limited" "-dwarf-version=4" "-debugger-tuning=gdb" "-D" "UA_ARCHITECTURE_WIN32" "-D" "UA_DYNAMIC_LINKING_EXPORT" "-Wall" "-Wextra" "-Wpedantic" "-Werror" "-Wno-static-in-inline" "-Wno-overlength-strings" "-Wno-unused-parameter" "-Wc++-compat" "-Wformat" "-Wformat-security" "-Wformat-nonliteral" "-Wmissing-prototypes" "-Wstrict-prototypes" "-Wredundant-decls" "-Wuninitialized" "-Winit-self" "-Wcast-qual" "-Wstrict-overflow" "-Wnested-externs" "-Wmultichar" "-Wundef" "-std=c99" "-ferror-limit" "19" "-fmessage-length" "0" "-fno-use-cxa-atexit" "-fgnuc-version=4.2.1" "-fobjc-runtime=gcc" "-fexceptions" "-fseh-exceptions" "-fdiagnostics-show-option" "-faddrsig" "-x" "c" "ua_pubsub_ns0-c752d2.c"

fancy@win-vm MSYS /c/Users/fancy/Downloads/PipelineFiles
$ ./ua_pubsub_ns0-c752d2.sh

fancy@win-vm MSYS /c/Users/fancy/Downloads/PipelineFiles
$ ./ua_pubsub_ns0-c752d2.sh
Stack dump:
0.      Program arguments: C:\msys64\mingw64\bin\clang.exe -cc1 -triple x86_64-w64-windows-gnu -emit-obj -mrelax-all -disable-free -disable-llvm-verifier -discard-value-names -main-file-name ua_pubsub_ns0.c -mrelocation-model pic -pic-level 2 -mthread-model posix -mframe-pointer=none -relaxed-aliasing -fmath-errno -fno-rounding-math -masm-verbose -mconstructor-aliases -munwind-tables -target-cpu x86-64 -dwarf-column-info -debug-info-kind=limited -dwarf-version=4 -debugger-tuning=gdb -D UA_ARCHITECTURE_WIN32 -D UA_DYNAMIC_LINKING_EXPORT -Wall -Wextra -Wpedantic -Werror -Wno-static-in-inline -Wno-overlength-strings -Wno-unused-parameter -Wc++-compat -Wformat -Wformat-security -Wformat-nonliteral -Wmissing-prototypes -Wstrict-prototypes -Wredundant-decls -Wuninitialized -Winit-self -Wcast-qual -Wstrict-overflow -Wnested-externs -Wmultichar -Wundef -std=c99 -ferror-limit 19 -fmessage-length 0 -fno-use-cxa-atexit -fgnuc-version=4.2.1 -fobjc-runtime=gcc -fexceptions -fseh-exceptions -fdiagnostics-show-option -faddrsig -x c ua_pubsub_ns0-c752d2.c
1.      <eof> parser at end of file
2.      Code generation
 #0 0x000000000278f4b8 (C:\msys64\mingw64\bin\clang.exe+0x238f4b8)
 #1 0x0000000002795e7c (C:\msys64\mingw64\bin\clang.exe+0x2395e7c)
 #2 0x00000000027a4002 (C:\msys64\mingw64\bin\clang.exe+0x23a4002)
 #3 0x00000000027a8378 (C:\msys64\mingw64\bin\clang.exe+0x23a8378)
 #4 0x000000000277f8d8 (C:\msys64\mingw64\bin\clang.exe+0x237f8d8)
 #5 0x000000000179c2bc (C:\msys64\mingw64\bin\clang.exe+0x139c2bc)
 #6 0x00000000017a58a0 (C:\msys64\mingw64\bin\clang.exe+0x13a58a0)
 #7 0x0000000001f83fc1 (C:\msys64\mingw64\bin\clang.exe+0x1b83fc1)
 #8 0x00000000047a1e09 (C:\msys64\mingw64\bin\clang.exe+0x43a1e09)
 #9 0x00000000033b7919 (C:\msys64\mingw64\bin\clang.exe+0x2fb7919)
#10 0x0000000002520569 (C:\msys64\mingw64\bin\clang.exe+0x2120569)
#11 0x00000000024e51d1 (C:\msys64\mingw64\bin\clang.exe+0x20e51d1)
#12 0x00000000025d3339 (C:\msys64\mingw64\bin\clang.exe+0x21d3339)
#13 0x000000000040269f (C:\msys64\mingw64\bin\clang.exe+0x269f)
#14 0x000000000040197a (C:\msys64\mingw64\bin\clang.exe+0x197a)
#15 0x0000000004a9b01e (C:\msys64\mingw64\bin\clang.exe+0x469b01e)
#16 0x00000000004013c4 (C:\msys64\mingw64\bin\clang.exe+0x13c4)
#17 0x000000000040151b (C:\msys64\mingw64\bin\clang.exe+0x151b)
#18 0x00007ffbdd537bd4 (C:\WINDOWS\System32\KERNEL32.DLL+0x17bd4)
#19 0x00007ffbdeb4ce51 (C:\WINDOWS\SYSTEM32\ntdll.dll+0x6ce51)
./ua_pubsub_ns0-c752d2.sh: Zeile 4:  1176 Segmentation fault      "C:\\msys64\\mingw64\\bin\\clang.exe" "-cc1" "-triple" "x86_64-w64-windows-gnu" "-emit-obj" "-mrelax-all" "-disable-free" "-disable-llvm-verifier" "-discard-value-names" "-main-file-name" "ua_pubsub_ns0.c" "-mrelocation-model" "pic" "-pic-level" "2" "-mthread-model" "posix" "-mframe-pointer=none" "-relaxed-aliasing" "-fmath-errno" "-fno-rounding-math" "-masm-verbose" "-mconstructor-aliases" "-munwind-tables" "-target-cpu" "x86-64" "-dwarf-column-info" "-debug-info-kind=limited" "-dwarf-version=4" "-debugger-tuning=gdb" "-D" "UA_ARCHITECTURE_WIN32" "-D" "UA_DYNAMIC_LINKING_EXPORT" "-Wall" "-Wextra" "-Wpedantic" "-Werror" "-Wno-static-in-inline" "-Wno-overlength-strings" "-Wno-unused-parameter" "-Wc++-compat" "-Wformat" "-Wformat-security" "-Wformat-nonliteral" "-Wmissing-prototypes" "-Wstrict-prototypes" "-Wredundant-decls" "-Wuninitialized" "-Winit-self" "-Wcast-qual" "-Wstrict-overflow" "-Wnested-externs" "-Wmultichar" "-Wundef" "-std=c99" "-ferror-limit" "19" "-fmessage-length" "0" "-fno-use-cxa-atexit" "-fgnuc-version=4.2.1" "-fobjc-runtime=gcc" "-fexceptions" "-fseh-exceptions" "-fdiagnostics-show-option" "-faddrsig" "-x" "c" "ua_pubsub_ns0-c752d2.c"

fancy@win-vm MSYS /c/Users/fancy/Downloads/PipelineFiles
$ ./ua_pubsub_ns0-c752d2.sh

fancy@win-vm MSYS /c/Users/fancy/Downloads/PipelineFiles
$ ./ua_pubsub_ns0-c752d2.sh

fancy@win-vm MSYS /c/Users/fancy/Downloads/PipelineFiles
$ 
```

**Comment by Pro** [comment]
Is it somehow possible to get the debug symbols for the clang.exe to at least see the method in the stacktrace where clang fails?

**Comment by Pro** [comment]
I tried to reproduce the issue on my linux machine with clang 10. There it doesn't occur, even with multiple runs and very limited memory.

So it is definitely an issue with the MingW version of Clang.

**Comment by mati865** [comment]
I looped it with `while true; do ./ua_pubsub_ns0-c752d2.sh; done` on Windows 10 machine and there was no crash after a few minutes.
MSYS2 doesn't modify the parser so it have to be an upstream issue. Could you run MSVC build with clang-cl?

> Is it somehow possible to get the debug symbols for the clang.exe to at least see the method in the stacktrace where clang fails?

You would have to rebuild Clang+LLVM yourself.

**Comment by Pro** [comment]
> I looped it with `while true; do ./ua_pubsub_ns0-c752d2.sh; done` on Windows 10 machine and there was no crash after a few minutes.

Ok, that's interesting. And you used a clean msys2 install as I mentioned above?
My VM has the following settings:

![image](https://user-images.githubusercontent.com/251973/84131436-d4a06180-aa44-11ea-85d7-c924e1c7881d.png)

Today evening I can try to reproduce it on my native Win machine.

> You would have to rebuild Clang+LLVM yourself.

Can you draft the necessary steps and flags to enable the debug build? (I'm not a mingw/llvm expert).
Maybe I have some time later to rebuild clang+llvm on the corresponding machine and get a better stack trace.

**Comment by mati865** [comment]
> Ok, that's interesting. And you used a clean msys2 install as I mentioned above?

I have extracted `msys2-base-x86_64-20200602.tar.xz`, upgraded it with `pacman -Syu` and then installed Clang with `pacman -S mingw-w64-x86_64-clang`.

> Can you draft the necessary steps and flags to enable the debug build?


Changing https://github.com/msys2/MINGW-packages/blob/b147ac6c6375bf065ac615987e8062cc7abf7ce1/mingw-w64-clang/PKGBUILD#L56
to
```
 options=('!debug' '!strip') 
```
should already give useful backtraces. `debug` is not tested much and I don't know if it would succeed.
Build with:
```
cd mingw-w64-clang
MINGW_INSTALLS=mingw64 makepkg-mingw -sf
```
Then install the package with `pacman -U <path_to_package>`.
Docs: https://www.msys2.org/wiki/Creating-Packages/#re-building-a-package

**Comment by Pro** [comment]
Ok, now I got the clang with debug symbols.
Unfortunately, with the current master version the issue does not appear.

What I did was cloning the current msys2/mingw-packages repo master (https://github.com/msys2/MINGW-packages/commit/d3e926661e02a86042f21cbdecdc019f1c8567e9) and did the build steps you mentioned.

If I replace the `clang.exe` with the original one, the issue reappears.

Here is the version output (I renamed the exe with debug symbols to `clang-debugsym.exe`):

```
user@vm MINGW64 /c/Users/user/Downloads/PipelineFiles
$ /C/msys64/mingw64/bin/clang.exe --version
clang version 10.0.0 (https://github.com/msys2/MINGW-packages.git 3f880aaba91a3d9cdfb222dc270274731a2119a9)
Target: x86_64-w64-windows-gnu
Thread model: posix
InstalledDir: C:\msys64\mingw64\bin

user@vm MINGW64 /c/Users/user/Downloads/PipelineFiles
$ /C/msys64/mingw64/bin/clang-debugsym.exe --version
clang version 10.0.0
Target: x86_64-w64-windows-gnu
Thread model: posix
InstalledDir: C:\msys64\mingw64\bin

```

To summarize, it seems `pacman -Syu` fixes the issue. When running it, it installs newer versions of llvm and clang. Also tested it already on azure.
